app/models/notification.rb

```diff
@@ -15,7 +15,7 @@ def self.for(recipient, opts={})
   end
 
   def self.notify(recipient, target, actor)
-    return nil unless target.respond_to? :notification_type
+    return nil unless target.respond_to? :notification_type || recipient != actor
```

Comment:
I'm afraid that's not changing anything, you got to retest.

``rb
[50] pry(main)> puts "foo" || "bar"
foo
=> nil
[51] pry(main)> puts("foo") || "bar"
foo
=> "bar"
```